geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala

```diff
@@ -164,13 +167,55 @@ class GeoMesaAccumuloInputFormat extends InputFormat[Text, SimpleFeature] with L
    *
    * Our delegated AccumuloInputFormat creates a split for each range - because we set a lot of ranges in
    * geomesa, that creates too many mappers. Instead, we try to group the ranges by tservers. We use the
-   * number of shards in the schema as a proxy for number of tservers.
+   * location of the tserver to determine the number of tservers.
    */
   override def getSplits(context: JobContext): java.util.List[InputSplit] = {
     init(context.getConfiguration)
     val accumuloSplits = delegate.getSplits(context)
-    logger.debug(s"Got ${accumuloSplits.length} splits")
-    accumuloSplits
+    // Get the appropriate number of mapper splits using the following priority
+    // 1. Get splits from AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS (geomesa.mapreduce.split.count.absolute)
+    // 2. Get splits from #tserver locations * AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER (geomesa.mapreduce.split.count.tablet)
+    // 3. Get splits from AccumuloInputFormat.getSplits(context)
+    val grpSplitsAbs: Option[Int] =
+      try {
+        val absSplits = AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS.get.toInt
+        if (absSplits > 0) Some(absSplits) else None
+      } catch {
+        case e: java.lang.NumberFormatException =>
+          logger.warn(s"Unable to parse geomesa.mapreduce.split.count.absolute = " +
+            s"${AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS.get} is not a valid Int.")
+          None
+      }
+
+    lazy val grpSplitsPerTServer: Option[Int] =
+      if (AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER.get == null) None
+      else {
+        val numLocations = accumuloSplits.flatMap(_.getLocations).distinct.head.length
+        val splitsPerTServer =
+          try {
+            AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER.get.toInt
+          } catch {
+            case e: java.lang.NumberFormatException =>
+              logger.warn(s"Unable to parse geomesa.mapreduce.split.count.tablet = " +
+                s"${AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER.get} is not a valid Int.")
```

Comment:
My previous note actually applies here. Though if you want this fixed here as well just to be clear that's fine.

It shouldn't, DESIRED_SPLITS_PER_TSERVER is a Geomesa system property which will default to null if it doesn't have a value. This error message is in an else statement of a null check on DESIRED_SPLITS_PER_TSERVER so at this point we know DESIRED_SPLITS_PER_TSERVER is not null and it has a value to get, it just can't be parsed as an Int.